filebeat/filebeat.full.yml

```diff
@@ -395,22 +395,206 @@ output.elasticsearch:
   # Configure curve types for ECDHE based cipher suites
   #tls.curve_types: []
 
+#------------------------------- Kafka output ---------------------------------
+#output.kafka:
+  # The list of Kafka broker addresses to connect to.
+  #hosts: ["localhost:9092"]
```

Comment:
tl;dr This list is only about fetching metadata, not about actual endpoints events are published too

To be more exact, it's the initial list of configured kafka brokers to fetch cluster metadata from in order to identify all kafka brokers and how topics/partitions are distributed between brokers. The actual connections to kafka brokers is made on advertised names being included in metadata.

Just giving an example this failing badly:
  - kafka broker in host `mykafka`
  - kafka broker advertises `localhost`
  - beats running on different machine with `hosts` being configured to `mykafka:9092`.

Having this setup, beat will connect to mykafka:9092 in order to fetch the metadata. The metadata state, there is one (and only one) kafka broker being available on localhost:9092 -> beats will fail to publish events due to no kafka instance running on `localhost:9200`.

If possible, it's recommended to configure multiple brokers (3 or 5 for example), in case the broker is down.